db/commitlog/commitlog.cc

```diff
@@ -1609,7 +1626,9 @@ future<> db::commitlog::segment_manager::delete_file(const sstring& filename) {
     return seastar::file_size(filename).then([this, filename](uint64_t size) {
         clogger.debug("Deleting segment file {}", filename);
         return commit_io_check(&seastar::remove_file, filename).then([this, size] {
-            totals.total_size_on_disk -= size;  
+            clogger.trace("Reclaimed {} MB", size/(1024*1024));
+            totals.total_size_on_disk -= size;
+            _disk_deletions.signal(size);
```

Comment:
The use of `_disk_deletions` look suspicious to me. We wait only when we exceed the limit, but always signal on delete. So if we delete while under the limit we may allocate when we over it.

Reply:
Yeah, you are right. Should replace with a shard future (or try to keep the semaphore synced with alloc/delete files - but that would complicate exception handling something fierce...)